### ISSUE TITLE
Breaking change - `Microsoft.StorageMover` resource provider registration required in provider v3.51.0

### ISSUE THREAD

**Issue by bubbletroubles** [hashicorp/terraform-provider-azurerm#21363]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.x

### AzureRM Provider Version

 v3.51.0

### Affected Resource(s)/Data Source(s)

any

### Terraform Configuration Files

```hcl
any resource
```

### Debug Output/Panic Output

```shell
│ Original Error: Cannnot register providers: Microsoft.StorageMover. Errors were: Cannot register provider Microsoft.StorageMover with Azure Resource Manager: resources.ProvidersClient#Register: Failure responding to request: StatusCode=403 
```

### Expected Behaviour

When deploying a config with the `azurerm_storage_mover` resource, and the resource provider is not registered, then a message appear to register the provider.

When deploying any config *without* the `azurerm_storage_mover` resource, that the Microsoft.StorageMover provider is not required.

### Actual Behaviour

The deployment fails because Microsoft.StorageMover is not registered in the subscriptyion, and the service connection deploying the resource does not have permission to register it.

### Steps to Reproduce

Deploy any AzureRM resource (except Storage Mover) to a subscription without Microsoft.StorageMover registered, using a service connection scoped at any level other than the subscription.
The deployment fails with the error

`│ Original Error: Cannnot register providers: Microsoft.StorageMover. Errors were: Cannot register provider Microsoft.StorageMover with Azure Resource Manager: resources.ProvidersClient#Register: Failure responding to request: StatusCode=403 `

### Important Factoids

Microsoft.StorageMover was added as a required resource provider in AzureRM 3.51.0. However, it's not a common resource, and is a breaking change.

[Microsoft's recommendation is to](https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/azure-services-resource-providers#registration) "Only register a resource provider when you're ready to use it. The registration step enables you to maintain least privileges within your subscription. A malicious user can't use resource providers that aren't registered.".  

The AzureRM provider requiring registration of a non-mainstream resource is not best practice. 

The release notes for the AzureRM provider release 3.51.0 do not mention that a new resource registration is required. 

### References

[Storage Mover Pull Request](https://github.com/hashicorp/terraform-provider-azurerm/pull/21000)
[Storage Mover in required resource providers](https://github.com/hashicorp/terraform-provider-azurerm/blob/e440014e535c0c2098e433076b50d81f0aa5008a/internal/resourceproviders/required.go#L73)

**Comment by ASHR4** [comment]
I'm having this as well, an interim fix is adding s`kip_provider_registration = "true"` to your provider block:

`provider "azurerm" {
  skip_provider_registration = "true"
  features {}
}`

Hope this gets fixed soon!

**Comment by ChrisTav424** [comment]
We are seeing this issue too. Thanks @ASHR4 we will try that

**Comment by sinbai** [comment]
@bubbletroubles thanks for opening this issue. PR #21370 to fix this issue has been merged. That fix is expected to be released in the upcoming release v3.52.0 of the Terraform provider , so could you please track that to see whether you get unlocked?

**Comment by C-Roche** [comment]
> 

@sinbai Do you have an expected release date for v3.52.0? This issue is preventing deployments for me, but fixing it on my side (changing to a previously working version, for example) will cause me issues with my pipeline. If I know when v3.52.0 will be released, it will help me decide on my course of action.

**Comment by sinbai** [comment]
@C-Roche Since I am not in charge of releases, I could not give an accurate release date for v3.52.0. But according to the release date [here](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.51.0/CHANGELOG.md), it could be estimated the next release date should be before this weekend. Hope this is useful for you.

**Comment by sinbai** [comment]
Just letting everyone know that this fix has been release in released in [v3.52.0 of the Terraform Provider](https://github.com/hashicorp/terraform-provider-azurerm/blob/v3.52.0/CHANGELOG.md).

**Comment by GraemeMeyerGT** [comment]
Just to confirm that #21370 in 3.52.0 seems to have resolved this for me.
